### PR TITLE
fix warmup by decoupling state from message receipt walk

### PR DESCRIPTION
**Pull request by vyzo**
this was accidentally broken in #6775.

### CODE REVIEW

**Review by Kubuxu — APPROVED** [review]
SGWM but could use tests

**Comment by vyzo** [comment]
Yeah, tests forthcoming soon -- we hav #6834 open.
I probably won't get to it today, as there are a couple of things ahead of it in the queue, but I'll get to it asap.

File: blockstore/splitstore/splitstore_compact.go [hunk 1]
```diff
@@ -631,14 +631,16 @@ func (s *SplitStore) endTxnProtect() {
 	s.txnMissing = nil
 }
 
-func (s *SplitStore) walkChain(ts *types.TipSet, inclState abi.ChainEpoch, inclMsgs abi.ChainEpoch,
+func (s *SplitStore) walkChain(ts *types.TipSet, inclState, inclMsgs, inclMsgReceipts abi.ChainEpoch,
```
Comment (magik6k) [inline blockstore/splitstore/splitstore_compact.go:634]: Why separate height for receipts?

Comment (vyzo) [reply blockstore/splitstore/splitstore_compact.go:634]: well, I wanted to decouple it from state as it was the source of the bug -- when we sync from snapshot we dont have message receipts at all!

We could ostensibly couple them with the messages, but I don't think we need them at all in order to support the use cases -- please correct me if I am wrong :)

Comment (vyzo) [reply blockstore/splitstore/splitstore_compact.go:634]: following sync discussion, we discussed to collate them with message retention.

**Review by magik6k — APPROVED** [review]

**Pull request merged**